menu/import_export/halo1_model.py

```diff
@@ -0,0 +1,124 @@
+import bpy
+from bpy.utils import register_class, unregister_class
+from bpy.props import BoolProperty, FloatProperty, StringProperty, EnumProperty
+from bpy_extras.io_utils import ImportHelper, ExportHelper, orientation_helper, path_reference_mode, axis_conversion
+
+from ...halo1.model import read_halo1model, import_halo1_nodes
+from ...constants import SCALE_MULTIPLIERS
+
+#@orientation_helper(axis_forward='-Z') Find the right value for this.
+class MT_krieg_ImportHalo1Model(bpy.types.Operator, ImportHelper):
+	"""
+	The import operator for gbxmodel/jms models.
+	This stores the properties when inside of the import dialog, and
+	it specifies what to show on the side panel when importing.
+	"""
+	bl_idname = "import_scene.halo1_model"
+	bl_label = "Import Halo 1 Model"
+	bl_options = {'PRESET', 'UNDO'}
+
+	# Import-file-dialog settings:
+
+	filename_ext = ".gbxmodel"
+	filter_glob: StringProperty(
+		default="*.gbxmodel;*.model;*.jms",
+		options={'HIDDEN'},
+	)
+
+	# Node settings:
+
+	use_nodes: BoolProperty(
+		name="Import Nodes",
+		description="Import the nodes/frames.",
+		default=True,
+	)
+	node_size: FloatProperty(
+		name="Node Scene Size",
+		description="Set the size that nodes should have in the Blender scene if at 1.0 scale.",
+		default=0.1,
+		min=0.0,
+	)
+	use_armatures: BoolProperty(
+		name="Import Nodes as Armature",
+		description="Import the nodes as armature.",
```

Comment:
What is an armiture? Is that like blender's version of a bone?

Reply:
Yeah, an armature is a set of bones in Blender basically. A very popular format for their rigs.